### PR TITLE
[22.x] fs: Make compatible with boost 1.78

### PR DESCRIPTION
**Pull request by fanquake**
Backports #24104 to the `22.x` branch to fix the build with Boost 1.78.0.

### CODE REVIEW

File: src/wallet/load.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ bool VerifyWallets(interfaces::Chain& chain)
         fs::path wallet_dir = gArgs.GetArg("-walletdir", "");
         boost::system::error_code error;
         // The canonical path cleans the path, preventing >1 Berkeley environment instances for the same directory
-        fs::path canonical_wallet_dir = fs::canonical(wallet_dir, error);
+        fs::path canonical_wallet_dir = fs::canonical(wallet_dir, error).remove_trailing_separator();
```
Comment (hebasto) [inline src/wallet/load.cpp:25]: Why this change is required? I see no broken tests on the current 22.x branch (9b5f674abb0e90280436fbacb8711250cde11773) on macOS 12.2.1 (Apple M1).

Comment (achow101) [reply src/wallet/load.cpp:25]: On my system with boost 1.78, the unit tests fail with a trailing backslash in the path here.

**Comment by achow101** [comment]
ACK 021c3d892fb6b801143bb433e85f329fd250c97d

**Review by hebasto — COMMENTED** [review]
ACK 021c3d892fb6b801143bb433e85f329fd250c97d

**Review by hebasto — APPROVED** [review]

**Comment by gruve-p** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/24476/commits/021c3d892fb6b801143bb433e85f329fd250c97d
